src/Migration/Sources/CSV.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+namespace Utopia\Migration\Sources;
+
+use Utopia\Database\Database as UtopiaDatabase;
+use Utopia\Migration\Exception;
+use Utopia\Migration\Resource;
+use Utopia\Migration\Resources\Database\Attribute;
+use Utopia\Migration\Resources\Database\Collection;
+use Utopia\Migration\Resources\Database\Database;
+use Utopia\Migration\Resources\Database\Document;
+use Utopia\Migration\Resources\Storage\File;
+use Utopia\Migration\Source;
+use Utopia\Migration\Sources\Appwrite\Reader;
+use Utopia\Migration\Sources\Appwrite\Reader\Database as DatabaseReader;
+use Utopia\Migration\Transfer;
+use Utopia\Storage\Device;
+
+class CSV extends Source
+{
+    private string $filePath;
+
+    /**
+     * format: `{databaseId:collectionId}`
+     */
+    private string $resourceId;
+
+    private Device $device;
+
+    private Reader $database;
+
+    public function __construct(
+        string $resourceId,
+        string $filePath,
+        Device $device,
+        ?UtopiaDatabase $dbForProject
+    ) {
+        $this->device = $device;
+        $this->filePath = $filePath;
+        $this->resourceId = $resourceId;
+        $this->database = new DatabaseReader($dbForProject);
+    }
+
+    public static function getName(): string
+    {
+        return 'CSV';
+    }
+
+    public static function getSupportedResources(): array
+    {
+        return [
+            Resource::TYPE_DOCUMENT,
+        ];
+    }
+
+    // called before the `exportGroupDatabases`.
+    public function report(array $resources = []): array
+    {
+        $report = [];
+
+        if (! $this->device->exists($this->filePath)) {
+            return $report;
+        }
+
+        $file = new \SplFileObject($this->filePath, 'r');
+        $file->setFlags(\SplFileObject::READ_CSV | \SplFileObject::SKIP_EMPTY);
+
+        $file->seek(PHP_INT_MAX);
+        $rowCount = $file->key();
+
+        // Subtract to exclude header
+        if ($rowCount > 0) {
+            $rowCount--;
+        }
```

Comment:
We need to subtract, it's a 0-based count. So if key == 1 then there's 1 non-header row

Reply:
Seems correct from what I tested with multiple files.
1. Full `$rowCount` = 101 for 100 rows + 1 header line
2. Full `$rowCount` = 10,001 for 10K rows + 1 header line

Correct row count after this subtraction -
  - 100
  - 10,000